x-pack/plugins/observability_solution/infra/server/routes/custom_dashboards/save_custom_dashboard.ts

```diff
@@ -53,12 +58,16 @@ export function initSaveCustomDashboardRoute(framework: KibanaFramework) {
 
       const savedCustomDashboard = await savedObjectsClient.update<InfraCustomDashboard>(
         INFRA_CUSTOM_DASHBOARDS_SAVED_OBJECT_TYPE,
-        customDashboards.saved_objects[0].id,
-        payload
+        payload.id,
+        {
+          assetType: request.params.assetType,
+          ...payload,
+        }
```

Comment:
are we passing `id` again here?

Reply:
This was for updating the dashboard before I refactored it to be a separate PUT endpoint